firefox-ios/Client/Coordinators/Browser/BrowserCoordinator.swift

```diff
@@ -194,11 +200,13 @@ class BrowserCoordinator: BaseCoordinator,
     // MARK: - Route handling
 
     override func canHandle(route: Route) -> Bool {
+        // Due to the new AppEventQueue, this should not be needed anymore. Adding a fatal log to confirm
+        // this use case can be cleaned up
         guard browserIsReady, !tabManager.isRestoringTabs else {
             let readyMessage = "browser is ready? \(browserIsReady)"
             let restoringMessage = "is restoring tabs? \(tabManager.isRestoringTabs)"
             logger.log("Could not handle route, \(readyMessage), \(restoringMessage)",
-                       level: .info,
+                       level: .fatal,
```

Comment:
Fatal so we see this in Sentry. In theory this should never be called.

Reply:
This gets called (I believe) fairly frequently during normal usage if the app tries to open a link while it's restoring tabs. I see this in Jeremy's logs quite a bit. I think that scenario is the reason we have the `savedRoute` property. We might want to double-check on this to avoid any excessive errors in Sentry. cc @OrlaM

Reply:
I was not seeing that call in other logs, hence why that change. I think the `savedRoute` should be removed, as I had added that before we had the `AppEventQueue` as a patch to timing problems. It's a mechanism that doubles that logic. If we're not ready to handle deeplink, we shouldn't go in the coordinator I think. If we need to make changes to make that happen, then we should investigate that.

Reply:
> I think the savedRoute should be removed

I enthusiastically support this 😄, assuming we can do it safely; I don't have much context on why it was originally added, but I think it's a bit confusing at this point and just adds more complexity to this code (which has already got a lot going on).

Reply:
Ok so I reverted this fatal log for now since it seems as Matt pointed out there's still some occurrences where this is needed 😢 But, this means we have some changes to make before we can safely remove the `savedRoute`. Here's my attempt at this (see PR update)